sdks/java/extensions/sql/src/test/java/org/apache/beam/sdk/extensions/sql/meta/provider/pubsub/PubsubTableProviderIT.java

```diff
@@ -177,6 +177,125 @@ public void testSQLSelectsPayloadContent() throws Exception {
     resultSignal.waitForSuccess(Duration.standardMinutes(5));
   }
 
+  @Test
+  public void testSQLSelectsArrayAttributes() throws Exception {
+    String createTableString =
+        String.format(
+            "CREATE EXTERNAL TABLE message (\n"
+                + "event_timestamp TIMESTAMP, \n"
+                + "attributes ARRAY<ROW<key VARCHAR, `value` VARCHAR>>, \n"
+                + "payload ROW< \n"
+                + "             id INTEGER, \n"
+                + "             name VARCHAR \n"
+                + "           > \n"
+                + ") \n"
+                + "TYPE '%s' \n"
+                + "LOCATION '%s' \n"
+                + "TBLPROPERTIES '{ "
+                + "%s"
+                + "\"timestampAttributeKey\" : \"ts\" }'",
+            tableProvider.getTableType(), eventsTopic.topicPath(), payloadFormatParam());
+
+    String queryString = "SELECT message.payload.id, attributes[0].key AS name FROM message";
```

Comment:
You want `attributes[1].key`. SQL arrays are 1-indexed.

Reply:
Good to know! I restructured this test to use 1 indexing.